packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -252,82 +194,32 @@ export class GarbageCollector implements IGarbageCollector {
 		);
 
 		let gcVersionInBaseSnapshot: number | undefined;
-
-		/**
-		 * Sweep timeout is the time after which unreferenced content can be swept.
-		 * Sweep timeout = session expiry timeout + snapshot cache expiry timeout + one day buffer.
-		 *
-		 * The snapshot cache expiry timeout cannot be known precisely but the upper bound is 5 days.
-		 * The buffer is added to account for any clock skew or other edge cases.
-		 * We use server timestamps throughout so the skew should be minimal but make it 1 day to be safe.
-		 */
-		function computeSweepTimeout(sessionExpiryTimeoutMs: number | undefined) {
-			const maxSnapshotCacheExpiryMs = 5 * oneDayMs;
-			const bufferMs = oneDayMs;
-			return (
-				sessionExpiryTimeoutMs &&
-				sessionExpiryTimeoutMs + maxSnapshotCacheExpiryMs + bufferMs
-			);
-		}
-
-		/**
-		 * The following GC state is enabled during container creation and cannot be changed throughout its lifetime:
-		 * 1. Whether running GC mark phase is allowed or not.
-		 * 2. Whether running GC sweep phase is allowed or not.
-		 * 3. Whether GC session expiry is enabled or not.
-		 * For existing containers, we get this information from the metadata blob of its summary.
-		 */
 		if (createParams.existing) {
-			gcVersionInBaseSnapshot = getGCVersion(metadata);
-			// Existing documents which did not have metadata blob or had GC disabled have version as 0. For all
-			// other existing documents, GC is enabled.
-			this.gcEnabled = gcVersionInBaseSnapshot > 0;
-			this.sweepEnabled = metadata?.sweepEnabled ?? false;
-			this.sessionExpiryTimeoutMs = metadata?.sessionExpiryTimeoutMs;
-			this.sweepTimeoutMs =
-				metadata?.sweepTimeoutMs ?? computeSweepTimeout(this.sessionExpiryTimeoutMs); // Backfill old documents that didn't persist this
-			this.persistedGcFeatureMatrix = metadata?.gcFeatureMatrix;
+			gcVersionInBaseSnapshot = getGCVersion(createParams.metadata);
```

Comment:
Move this to config rather than living here and passing it in?

Reply:
done